versioned/storage/common-tests/src/main/java/org/projectnessie/versioned/storage/commontests/AbstractReferenceLogicTests.java

```diff
@@ -76,6 +85,59 @@ public class AbstractReferenceLogicTests {
 
   @NessiePersist protected Persist persist;
 
+  @SuppressWarnings("BusyWait")
+  @RepeatedTest(2)
+  public void refCreationDeletionWithConcurrentRefsListing() throws Exception {
+    IntFunction<String> refName = i -> "refs/heads/branch-" + i + "-" + (i & 7);
+
+    int num = 100;
+
+    List<Integer> indexes = new ArrayList<>(num);
+    for (int i = 0; i < num; i++) {
+      indexes.add(i);
+    }
+    ReferenceLogic refLogic = referenceLogic(persist);
+
+    @SuppressWarnings("resource")
+    ExecutorService exec = newSingleThreadExecutor();
+    try {
+      AtomicBoolean stop = new AtomicBoolean();
+      Future<?> task =
+          exec.submit(
+              () -> {
+                while (!currentThread().isInterrupted() && !stop.get()) {
+                  refLogic
+                      .queryReferences(referencesQuery("refs/heads/"))
+                      .forEachRemaining(x -> {});
+                  try {
+                    sleep(1);
+                  } catch (InterruptedException e) {
+                    break;
+                  }
+                }
+              });
+      try {
+        for (int i = 0; i < num; i++) {
+          refLogic.createReference(refName.apply(i), EMPTY_OBJ_ID);
+          sleep(1);
+        }
+        Collections.shuffle(indexes);
```

Comment:
Do we have to have a long list of references for this test? Why not operate on one reference? I suppose the test aims to cause a read/delete collision on one particular reference, so having a smaller list should actually increase the likelihood of collisions, I guess :thinking:

Reply:
Yea, that's actually correct. I used this test originally to reproduce the issue reported on the Nessie chat.